### DISCUSSION TITLE
Rename\transform a key\selector

### DISCUSSION THREAD

**Discussion by electriquo** [TomWright/dasel#240] [Q&A] [answered]
Is it possible to rename\transform a key\selector? if so, how?

E.g., having the following JSON

```json
{ "foo": "spam" }
```

How can `foo` be changed to `bar` to get?

```json
{ "bar": "spam" }
```

**Comment by TomWright** [comment, marked as answer]
There isn't a specific rename feature but you could do something like this:
```bash
$ echo '{"foo":"spam"}' > file.json
dasel put string -f file.json '.bar' $(dasel -f file.json --plain '.foo')
dasel delete -f file.json '.foo'
~ [] $ cat file.json
{
  "bar": "spam"
}
```

**Comment by myersjustinc** [comment]
I'd love to see some sort of feature along these lines as well. I ran into this today and was hoping to use dasel for it since I learned about it yesterday.

I couldn't figure it out from the docs, and rather than `put` and `delete` with a temporary file as above (since I saw that solution after the fact), I ended up piping through both [jq's `map`](https://stedolan.github.io/jq/manual/#map%28x%29%2Cmap%5Fvalues%28x%29) (one of my favorite aspects of that syntax) for the renaming and [csvkit's `in2csv`](https://csvkit.readthedocs.io/en/latest/scripts/in2csv.html) for the format conversion:

```
$ echo '[["foo@example.com","Foo","Bar"],["ham@eggs.example","Ham","and Eggs"]]' | \
    jq 'map({"Email": .[0], "First name": .[1], "Last name": .[2]})' | \
    in2csv --format json
Email,First name,Last name
foo@example.com,Foo,Bar
ham@eggs.example,Ham,and Eggs
```

**Reply by TomWright** [reply]
The correct solution right now would be to pipe a `put` and a `delete` together, but maybe a `map` command may be in order.

**Reply by TomWright** [reply]
Hey @myersjustinc,

With dasel v2 (still in development but in a good place) you can do this with:

```bash
echo '[["foo@example.com","Foo","Bar"],["ham@eggs.example","Ham","and Eggs"]]' 
| dasel -r json -w csv 'all().mapOf(email,[0],first,[1],last,[2]).merge()'
email,first,last
foo@example.com,Foo,Bar
ham@eggs.example,Ham,and Eggs
```

If you're interested in using the v2 development version you can find the docs here: https://daseldocs.tomwright.me/v/v2/
